base/ryu/Ryu.jl

```diff
@@ -90,10 +87,13 @@ function Base.show(io::IO, x::T) where {T <: Base.IEEEFloat}
         x = round(x, sigdigits=6)
     end
     buf = Base.StringVector(neededdigits(T))
```

Comment:
Random suggestion: since this is already structured to create a `String`, we could add a method to `string(x)` that avoids the final write call that makes an extra copy.